### PR TITLE
add in braintree transaction

### PR DESCRIPTION
**Pull request by andymckay**
* fixes mozilla/payments#74 
* makes the Namespaced object simpler and a bit more flexible
* add in braintree transaction endpoint
* catch the submitted for settlement webhook as well

* requires #472, so maybe read that first

### CODE REVIEW

File: docs/topics/braintree.rst [hunk 1]
```diff
@@ -281,7 +281,133 @@ which then passes it on to this endpoint. For more information see the
...
+.. _braintree-transaction-label:
+
+Transaction
+-----------
+
+The webhook returns transaction details to solitude. Solitude then creates a
+generic transaction object. It also creates a Braintree transaction that
+contains some information about the transaction.
+
+.. http:get:: /braintree/mozilla/subscription/<subscription id>/
```
Comment (kumar303) [inline docs/topics/braintree.rst]: I'm confused. This was meant to be ``transaction``, right?

File: lib/brains/serializers.py [hunk 2]
```diff
@@ -3,28 +3,30 @@
...
     @property
     def data(self):
-        return {
-            'mozilla': self.mozilla.data,
-            'braintree': self.braintree.data
-        }
+        def traverse(d):
+            for k, v in d.iteritems():
+                if isinstance(v, dict):
+                    traverse(v)
```
Comment (kumar303) [inline lib/brains/serializers.py]: ooh, fun times

File: lib/brains/serializers.py [hunk 3]
```diff
@@ -66,6 +68,26 @@ def resource_uri(self, pk):
...
+
+    class Meta:
+        model = BraintreeTransaction
+        read_only_fields = (
+            'billing_period_end_date', 'billing_period_start_date', 'kind',
+            'next_billing_date', 'next_billing_period_amount'
+        )
+
+    def resource_uri(self, pk):
+        return reverse('generic:transaction-detail',
```
Comment (kumar303) [inline lib/brains/serializers.py]: this URL doesn't look correct

**Comment by kumar303** [comment]
r+wc

File: lib/brains/serializers.py (hunk 3 above)
Comment (andymckay) [inline lib/brains/serializers.py]: well spotted

**Pull request merged**